app/models/canonical/craftables_crafting_material.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Canonical
+  class CraftablesCraftingMaterial < ApplicationRecord
+    self.table_name = 'canonical_craftables_crafting_materials'
+
+    belongs_to :craftable, polymorphic: true
+    belongs_to :material, class_name: 'Canonical::Material'
+
+    # rubocop:disable Lint/RedundantCopDisableDirective
+    # rubocop:disable Rails/UniqueValidationWithoutIndex
```

Comment:
So this is fucking weird and I hate it, but there was a weird disparity between local and CI Rubocop behaviour where, if I didn't disable `Rails/UniqueValidationWithoutIndex`, the cop would fail locally even though the unique index exists (and is created in exactly the same way as the index on  the `canonical_enchantables_enchantments` table, which doesn't produce the same failure). However, in CI, Rubocop determined that this failure would not occur and that the disablement was, therefore, redundant. Thus, I had to also disable that cop, and I had to do it in a separate comment before the one that was deemed redundant.

I tried without success to determine the reason for the disparity between the two models, as well as that for the disparity between local and CI environments. In the pre-commit hook, we do use the command line Rubocop tool rather than the Rails rake task that is used in CI. However, running the Rake task locally results not only in the same failure as the precommit hook finds, but also other failures that don't occur either in CI or in the precommit hook. I'm stumped.

Reply:
I found that, for some reason, although migrations weren't raising any errors, the composite index specified in the migration file was not actually created in the schema. Rolling back the migration and trying again seemed to update the schema so the Rubocop disablement is no longer necessary :tada: